### ISSUE TITLE
[depth_image_proc] [ROS2] symbol lookup error

### ISSUE THREAD

**Issue by Kettenhoax** [ros-perception/image_pipeline#696]
When I build depth_image_proc version 6b649d2b592d6c07461f9232f86e186017fb4442 in galactic, and run the `depth_image_proc::PointCloudXyzNode` component, I get a symbol lookup error:

```bash
/opt/ros/galactic/lib/rclcpp_components/component_container: symbol lookup error: ../lib/libdepth_image_proc.so: undefined symbol: _ZN16depth_image_proc12convertDepthIfEEvRKSt10shared_ptrIKN11sensor_msgs3msg6Image_ISaIvEEEERS1_INS3_12PointCloud2_IS5_EEERKN14image_geometry18PinholeCameraModelEd
```

It seems that 635bbb94a7ca72be11bb914a736b5052a6ec0b8d introduced changes in which the definition of the convertDepth function, which seems to be the missing symbol, is moved to a cpp file. PointCloudXyzNode only includes depth_image_proc/conversions.hpp.
Am I missing something, or is the template never instantiated when building PointCloudXyzNode, and the symbol never generated?

**Comment by shrini-mm** [comment]
I am facing the same problem. Is there any update on this?

**Comment by dcconner** [comment]
Ditto with Foxy build

**Comment by shrini-mm** [comment]
I fixed it by installing the apt package of `depth_image_proc` that came with  ubuntu 20.04. But I am not sure how to fix it yet by doing our own build. Hope it helps.

**Comment by quentingllmt** [comment]
For those looking, applying https://github.com/ros-perception/image_pipeline/pull/718 fixed the runtime linker error for me on the ros2 branch.

**Comment by flynneva** [comment]
@Kettenhoax @shrini-mm @dcconner this should now be covered with #757 backporting the fix from #718. Sorry for causing so much trouble for everyone 😓 

fyi @JWhitleyWork

**Comment by JWhitleyWork** [comment]
Should be fixed on `ros2` branch now. Don't worry about it, @flynneva - it happens to everyone!
